connectors/slack/template/index.js

```diff
@@ -1,24 +1,295 @@
+const Url = require('url');
 const configure = require('./configure');
 const Sdk = require('@fusebit/add-on-sdk');
+const { verifyRequestSignature } = require('@slack/events-api');
+const storage = require('./storage');
+const Superagent = require('superagent');
 
-/*
+/**
+ * @param ctx {FusebitContext}
+ */
+module.exports = (ctx, cb) => {
+  let url = Url.parse(ctx.url);
+  let pathname = url.pathname.split('/');
+  pathname.splice(0, 5);
+  pathname = '/' + pathname.join('/');
+  url.pathname = pathname;
+  url = Url.format(url);
 
-Add-on Settings Manager contract: 
+  Sdk.debug('NEW REQUEST', ctx.method, ctx.url);
 
-Initialization - GET with query params
-- [required] returnTo (returnTo, required)
-- [optional] state (opaque, to roundtrip)
-- [optional] data (base64 encoded JSON object, to augment and roundtrip)
+  // Most requests go to the Settings Manager
+  if (!pathname.startsWith('/events')) {
+    Sdk.createSettingsManager(configure)(ctx).then(
+      function (result) {
+        cb(null, result);
+      },
+      function (err) {
+        cb(err, null);
+      }
+    );
+  } else {
+    if (ctx.configuration.slack_signing_secret) {
+      if (pathname.startsWith('/events/ingest') && ctx.method === 'POST') {
+        handleEventIngest(ctx, cb);
+      } else if (pathname.startsWith('/events/registration') && ctx.method === 'POST') {
+        return registerCore(ctx, cb);
+      } else if (pathname.startsWith('/events/registration') && ctx.method === 'DELETE') {
+        return registerCore(ctx, cb, true);
+        // COMMENT OUT WHEN NOT DEBUGGING
+        // } else if(pathname.startsWith('/events/storage') && ctx.method === 'GET') {
+        //     storage.ensureStorage(ctx);
+        //     storage.ensureCache().then(()=>{
+        //         cb(null, {
+        //             status: 200,
+        //             body: {
+        //                 tokenToTeam : storage.tokenToTeam,
+        //                 teamToHandler: storage.teamToHandler
+        //             }
+        //         });
+        //         return;
+        //     });
+        // } else if(pathname.startsWith('/events/storage') && ctx.method === 'DELETE'){
+        //     storage.ensureStorage(ctx);
+        //     storage.ensureCache().then(()=>{
+        //         storage.put(()=>{
+        //             storage.tokenToTeam = {};
+        //             storage.teamToHandler = {};
+        //         }).then(()=>{
+        //             cb(null, { status: 200 });
+        //             return;
+        //         });
+        //     });
```

Comment:
Do you want to remove before publishing?

Reply:
It's damn useful for debugging, what do you suggest I do with it? I already rewrote it like 3 times after removing

Reply:
Leave it in I guess.